torch/distributed/fsdp/flat_param.py

```diff
@@ -112,7 +112,7 @@ class HandleConfig:
     offload_params: bool
     low_prec_param_dtype: Optional[torch.dtype]
     low_prec_reduce_dtype: Optional[torch.dtype]
-    keep_low_precision_grads: bool = False
+    keep_low_precision_grads: bool
```

Comment:
looks OK to me, but curious, any reason for this change?

Reply:
IMHO, it is a good practice to not have defaults for internal methods and classes since defaults can lead to silent bugs from logical errors / mis-intention. The only tradeoff is verbosity.

For example, in this case, we added the `keep_low_precision_grads` attribute to `HandleConfig` later. By having this as defaulted to `False`, it may be convenient for backward compatibility, but it is also possible that an existing usage accidentally uses `False` only because we forgot to update the usage site appropriately. If we do not default, then we will have errors in every usage site that we did not update, and we are sure to check that we are passing the correct value in each case.